src/main/java/Duke.java

```diff
@@ -1,10 +1,186 @@
+                            Task taskDeadline = new Deadline(description, date);
+                            tasks.add(taskDeadline);
+                            printTask(taskDeadline, tasks);
+                            writeTaskToFile(filePath);
+                        } catch (ArrayIndexOutOfBoundsException e) {
+                            throw new DukeException("-------------------------------------------------------------------\n" +
+                                    "☹ OOPS!!! The deadline date cannot be empty.\n" +
+                                    "-------------------------------------------------------------------\n");
+                        }
+                    } catch (ArrayIndexOutOfBoundsException e) {
+                        throw new DukeException("-------------------------------------------------------------------\n" +
+                                "☹ OOPS!!! The deadline description cannot be empty.\n" +
+                                "-------------------------------------------------------------------\n");
+                    }
+                } else if (line.startsWith("event")) {
+                    try {
+                        String event = line.split(" ", 2)[1];
+                        try {
+                            String time = event.split("/")[1];
+                            String description = event.split(" /")[0];
+                            Task taskEvent = new Event(description, time);
+                            tasks.add(taskEvent);
+                            printTask(taskEvent, tasks);
+                            writeTaskToFile(filePath);
+                        } catch (ArrayIndexOutOfBoundsException e) {
+                            throw new DukeException("-------------------------------------------------------------------\n" +
+                                    "☹ OOPS!!! The event time cannot be empty.\n" +
+                                    "-------------------------------------------------------------------\n");
+                        }
+                    } catch(ArrayIndexOutOfBoundsException e) {
+                            throw new DukeException("-------------------------------------------------------------------\n" +
+                                    "☹ OOPS!!! The event description cannot be empty.\n" +
+                                    "-------------------------------------------------------------------\n");
+                    }
+                } else {
+                    throw new DukeException("-------------------------------------------------------------------\n" +
+                            "☹ OOPS!!! I'm sorry, but I don't know what that means :-(\n" +
+                            "-------------------------------------------------------------------\n");
+                }
+            } catch(DukeException e) {
+                System.out.println(e.getMessage());
+            }
+        }
```

Comment:
Consider refactoring this. It is a bit too bloated for one conditional.
Try applying DRY principles and some abstraction.
Mainly, the issue seem to be your main is handling both business logic and parsing.
Also try to clean up the nesting of your exceptions.
I see that it seems to be due to trying to display a different message for the same exception.
Perhaps it would be wiser for the business logic to throw a different exception for each step where the error can occur? 
This way, you can catch the errors in one block and handle them different rather than in nested ones.